[Chiew Chun Jia] iP

chewycj opened the pull request.

chewycj committed:
Rename chatbot, Add greet and exit

chewycj committed:
Add Main.java and implement echo and exit functions

chewycj committed:
Add Todo class and implement task list function

Any user input will be added to a task list, except the input 'list'.
When the user inputs 'list', Appal will list out current tasks

chewycj committed:
Rename Todo class to Task and add mark tasks function

Users can mark a task as done with the input "mark <task number>", and
unmark the task with the input "unmark <task number>"

chewycj committed:
Adjust layout to follow coding standard

chewycj committed:
Task class: Add subclasses ToDo, Deadline, Event

chewycj committed:
Appal.java: Improve readability in handleInput()

Extracted the welcome message, as the code fragment can be grouped
together.

Using switch() along with constants instead of if-else control flow
makes code more understandable, for example to understand valid command
inputs.

chewycj committed:
Add Parser class and remove magic literals

Adding the Parser class helps to extract details from the user input,
improving code readability in the Appal.java class.

Removing magic literals through through the use of constants also
improves code quality, by defining the values used in words.

Switch statements are also in-line to follow coding standards.

chewycj committed:
Modify files for automated testing

Trying out input output redirection for one user input

MatchaRRR commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,32 @@
+        int wordCount = words.length;
+
+        commandDetails[COMMAND_INDEX] = words[COMMAND_INDEX];
+        int commandDetailsIndex = COMMAND_INDEX + 1;
```
If COMMAND_INDEX is considered as a const, maybe commandDetailsIndex can also be a const COMMAND_DETAILS_INDEX.

MatchaRRR commented on src/main/java/Appal.java:
```diff
@@ -11,6 +12,20 @@ public class Appal {
+    public static final String UNKNOWN_INPUT_NOTICE = "Oops! I don't recognise this command :(";
+    public static final String TASK_DONE_MESSAGE = "Task done! One more step towards success :)";
+    public static final String UNMARK_TASK_MESSAGE = "What's next on the agenda? :D";
+
```
I like this constant initialization. It looks very neat.

MatchaRRR commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,32 @@
+        }
+        return commandDetails;
+    }
+}
```
It looks messy here, maybe you can make it more readable and nerdy?

chewycj replied:
I agree that the Parser code looks messy. I will try to edit it according to the coding standards!

MatchaRRR commented on src/main/java/Main.java:
```diff
@@ -1,4 +1,5 @@
 public class Main {
+    // Entrypoint to Appal
     public static void main(String[] args) {
         Appal appal = new Appal();
```
I like the idea of adding a Main

MatchaRRR reviewed (commented):
It looks nice. Some small changes needed.

wx-03 commented on src/main/java/Appal.java:
```diff
@@ -0,0 +1,154 @@
+
+    public void exitAppal() {
+        isExited = true;
+        printSeparator();
+        System.out.println(BYE_MESSAGE);
+        printSeparator();
```
These three lines can be extracted into a separate method `printReply(String reply)`
since they appear frequently in other parts of the code.

chewycj replied:
That's true! I will make the changes accordingly, thank you for the suggestion :)

wx-03 commented on src/main/java/Appal.java:
```diff
@@ -0,0 +1,154 @@
+            break;
+        case COMMAND_UNMARK:
+            markTask(commandDetails, false);
+            break;
```
I like how you extracted everything into separate methods. The code is very readable.

chewycj committed:
Appal.java and Parser.java: Improve code readability

Appal.java: Added more named constants, as well as abstracted out the
method printMessage() as it appears quite often in the code.

Parser.java: Added a named constant to make the start of the code more
understandable, and added more details of how the Parser extracts
parameters in user input in the comments.

chewycj committed:
Improve code readability and aesthetic

Added a named constant for the value 100 to indicate the maximum number
of tasks to be stores, as well as renamed some variables to make code
more understandable.

Added an Appal logo to the welcomeUser() function.

chewycj committed:
Implement exceptions for user input

Exceptions are specific to each command: todo, event, deadline, and
mark. These exceptions are subclasses of the superclass AppalException,
which checks for generic invalid input.

Appal.java is edited such that methods can throw these exceptions when
necessary.

chewycj committed:
Bug fix: Add constructor method that takes in a string argument

This constructor method is needed for other subclasses of the
AppalException class to print a specific message.

chewycj committed:
Appal.java: Refactor code

There is a need to check if the todo, deadline, and event commands have
a specified task, hence the chunk of code is extracted out as a method
from the adTodo(), addDeadline() and addEvent() methods.

chewycj committed:
Appal.java: Add exception and update named constants and handleInput()

Added the ArrayIndexOutOfBoundsException to markTask() as any input that
is less than or equals to zero, or more than 100, will result in this
exception.

Some named constants were updated to make the chatbot interface look
more consistent.

In handleInput(), instead of creating a new Scanner object each time we
take in user input, a Scanner object is instantiated as an attribute of
the Appal class.

chewycj committed:
Update files for automated testing

chewycj committed:
Merge branch 'branch-Level-5'

chewycj committed:
Divide classes into packages

Organize the classes into suitable java packages.

chewycj committed:
Merge branch 'branch-A-Packages'

chewycj committed:
Implement taskList as an ArrayList instead of an array

Used Java Collections classes for storing task data instead, as it
offer many advantages (e.g., dynamic sizing, easy to find/add/delete
items), over using a primitive data structure such as a normal array.

chewycj committed:
Change exception type caught in markTask()

Due to the change of type of the taskList from array to ArrayList, the
type of exception caught changed from ArrayIndexOutOfBoundsException to
IndexOutOfBoundsException.

chewycj committed:
Implement delete task function

chewycj committed:
Implement function to load saved tasks from a text file

chewycj committed:
Implement function to write current tasks to a text file

Added formatting functions in the Task class and Task subclasses.

chewycj committed:
Appal.java: Add message for successful loadExistingTasksData()

chewycj committed:
Add exceptions to handle I/O errors

chewycj committed:
Merge branch 'branch-Level-6'

chewycj committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/appal/Appal.java
#	src/main/java/appal/task/Task.java

chewycj committed:
Appal.java: Improve code quality based on coding standards

Added named constants and line wrapping.

chewycj committed:
Appal.java: Refactor code

chewycj committed:
Create JAR manifest

chewycj committed:
Add Ui class and Message class

Ui class handles user interface messages and inputs, while message class
stores message constants

okkhoy commented on src/main/java/appal/Appal.java:
```diff
@@ -0,0 +1,282 @@
+package appal;
+
+import appal.exception.*;
```
don't use wild card imports

okkhoy commented on src/main/java/appal/Appal.java:
```diff
@@ -0,0 +1,282 @@
+import java.util.ArrayList;
+
+public class Appal {
+    // Constants for commands
```
Consider creating a separate class for these constants

okkhoy commented on src/main/java/appal/Appal.java:
```diff
@@ -0,0 +1,282 @@
+            handleInput(inputDetails, true);
+        }
+    }
+}
```
This class seems to be doing a lot; consider extracting related methods into a separate class (e.g., individual command handling)

okkhoy commented on src/main/java/appal/Appal.java:
```diff
@@ -0,0 +1,282 @@
+        }
+    }
+
+    public void handleInput(String[] inputDetails, boolean isFromUser) {
```
all non-trivial methods need header comments

okkhoy reviewed (commented):
In general, well written.
Consider refactoring the Appal class.

chewycj committed:
Add new OOP classes to Appal

chewycj committed:
Remove wildcard imports and use Utils class to store constants

chewycj committed:
Merge branch 'branch-A-MoreOOP'

chewycj committed:
Modify files for automated testing

chewycj committed:
Change Deadline class to take in deadline date as LocalDate

chewycj committed:
Merge pull request #1 from chewycj/branch-Level-8

Change Deadline class to take in deadline date as LocalDate

chewycj committed:
Implement Find function of Appal

chewycj committed:
Merge pull request #2 from chewycj/branch-Level-9

Implement Find function of Appal

chewycj committed:
Add JavaDoc comments for Main and Appal classes

chewycj committed:
Add JavaDoc comments to Ui class and update comments in Main class

chewycj committed:
Add JavaDoc comments to the task package

chewycj committed:
Add JavaDoc comments for storage package

chewycj committed:
Add JavaDoc comments for parser package

chewycj committed:
Add JavaDoc comments for exception package

chewycj committed:
Add JavaDoc comments for commands package

chewycj committed:
Merge pull request #3 from chewycj/branch-A-JavaDoc

Add JavaDoc comments to code

chewycj committed:
Implement function to check deadlines

chewycj committed:
Merge branch 'master' into branch-Level-8-Stretch

# Conflicts:
#	src/main/java/appal/commands/FindCommand.java
#	src/main/java/appal/task/Deadline.java

chewycj committed:
Merge pull request #4 from chewycj/branch-Level-8-Stretch

Implement function to check deadlines

chewycj committed:
Edit format of some JavaDoc comments

chewycj committed:
Remove wildcard imports

chewycj committed:
Add functionality to save tasks after changes in task list

chewycj committed:
Edit line wrapping in Messages class

chewycj committed:
Update docs/README.md

chewycj committed:
Edit features section in docs/README.md

chewycj committed:
Bug fix: Task count will not increment with invalid deadline

chewycj committed:
Modify files for automated testing
